jbmc/unit/java_bytecode/java_bytecode_parser/parse_inner_class.cpp

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************\
+
+ Module: Unit tests for converting inner classes
+
+ Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <java-testing-utils/load_java_class.h>
+#include <java_bytecode/java_types.h>
+#include <testing-utils/catch.hpp>
+
+SCENARIO(
+  "Parsing inner java classes",
+  "[core][java_bytecode][java_bytecode_parser]")
+{
+  GIVEN("A class file of an inner class.")
+  {
+    WHEN("Parsing an inner class called \"Inner\".")
+    {
+      const symbol_tablet &symbol_table = load_java_class(
+        "Outer$Inner", "./java_bytecode/java_bytecode_parser");
```

Comment:
:question: What happens if you load from `Outer` rather than starting on the inner class?